### PR TITLE
Enable edge to edge support for End of Year

### PR DESCRIPTION
**Pull request by MiSikora**
## Description

This adds edge to edge support and immersive mode to the EoY feature.

## Testing Instructions

1. Check if all stories are still displayed correctly.
2. Test screenshot sharing.
3. Test [upsell flow](https://github.com/Automattic/pocket-casts-android/pull/3066). Plus Interstitial story should be gone after a purchase is made.
4. Check support on a tablet.

### Edge to edge

https://github.com/user-attachments/assets/57cd012d-aafd-4c5c-84c6-3af36c6b4ad0

### No edge to edge

https://github.com/user-attachments/assets/caf32ebe-d353-4004-b221-31956da72900

## Checklist
- [ ] ~If this is a user-facing change, I have added an entry in CHANGELOG.md~
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] ~I have considered whether it makes sense to add tests for my changes~
- [ ] ~All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`~
- [ ] ~Any jetpack compose components I added or changed are covered by compose previews~
- [ ] ~I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.~

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt [hunk 1]
```diff
@@ -747,8 +747,9 @@ class MainActivity :
     }
 
     private fun showStories(source: StoriesSource) {
-        StoriesFragment.newInstance(source)
-            .show(supportFragmentManager, "stories_dialog")
+        StoriesActivity.open(this, source)
+//        StoriesFragment.newInstance(source)
+//            .show(supportFragmentManager, "stories_dialog")
```
Comment (MiSikora) [inline app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt:752]:
```suggestion
        StoriesActivity.open(this, source)
```

**Review by mebarbosa — APPROVED** [review]
LGTM

Just to double check, is this the expected behavior for tablet, right?

[Screen_recording_20241027_110820.webm](https://github.com/user-attachments/assets/c7002bda-fb39-4c43-8365-f6575bc7dd4a)

**Comment by MiSikora** [comment]
No, tapping outside should close the flow. I'll fix it.

**Pull request merged**